Add multi-targeting sections to trimming and AOT documentation (#50076)
dotnet/docs: docs/core/deploying/native-aot/index.md

These analyzers help to ensure that a library is compatible with Native AOT.

### Multi-targeting for AOT compatibility

> When preparing libraries for AOT compatibility, if your library targets any framework earlier than `net6.0` (such as `netstandard2.0` or `net472`), you should also multi-target to `net6.0` to ensure that apps targeting `net6.0` or above get a version of your library that supports the AOT analyzers. Use the `IsTargetFrameworkCompatible` MSBuild function to conditionally enable `IsAotCompatible` for `net6.0` and above:

Reviewer: AOT compat analysis was introduced for net8.0, so the AOT example and text should reference net8.0 instead of net6.0.

> When preparing libraries for AOT compatibility, if your library targets any framework earlier than `net8.0` (such as `netstandard2.0`, or `net472`), multi-target to `net8.0`. This ensures that apps targeting `net8.0` or above get a version of your library that supports the AOT analyzers. Also, include the latest .NET version so your library is analyzed with the latest analyzer. Use the `IsTargetFrameworkCompatible` MSBuild function to conditionally enable `IsAotCompatible` for `net8.0` and above:

Reviewer 2: This is a bit of a nit, but IIRC the main benefit is that multitargeting enables the AOT analyzers in the build of the library. `This ensures that apps targeting `net8.0` or above get a version of your library that supports the AOT analyzers.` to me reads like the main benefit is for the apps that use the library.

Something like this seems a bit clearer to me:

```suggestion
When preparing libraries for AOT compatibility, if your library targets any framework earlier than `net8.0` (such as `netstandard2.0`, or `net472`), multi-target to `net8.0`. The analyzers are not supported for compilations targeting earlier frameworks. Targeting `net8.0` and above will ensure the analyzers run on your library. Also, include the latest .NET version so your library is analyzed with the latest analyzer. Use the `IsTargetFrameworkCompatible` MSBuild function to conditionally enable `IsAotCompatible` for `net8.0` and above:
```
